test_unstructured/file_utils/test_filetype.py

```diff
@@ -85,13 +85,42 @@ def test_detect_filetype_from_filename(file_name: str, expected_value: FileType)
         ("fake-incomplete-json.txt", FileType.TXT),
     ],
 )
-def test_detect_filetype_from_filename_with_extension(
+def test_detect_filetype_from_filename_when_libmagic_not_available(
     file_name: str, expected_value: FileType, monkeypatch: MonkeyPatch
 ):
-    """File-type is detected from extension when libmagic not available or file does not exist."""
+    """File-type is detected using `filetype` library when libmagic is not available."""
     # -- when libmagic is not available --
     monkeypatch.setattr(filetype, "LIBMAGIC_AVAILABLE", False)
     assert detect_filetype(example_doc_path(file_name)) == expected_value
+
+
+@pytest.mark.xfail(reason="Disputed Behavior", raises=FileNotFoundError, strict=True)
+@pytest.mark.parametrize(
+    ("file_name", "expected_value"),
+    [
+        ("pdf/layout-parser-paper-fast.pdf", FileType.PDF),
+        ("fake.docx", FileType.DOCX),
+        ("img/example.jpg", FileType.JPG),
+        ("fake-text.txt", FileType.TXT),
+        ("eml/fake-email.eml", FileType.EML),
+        ("factbook.xml", FileType.XML),
+        ("example-10k.html", FileType.HTML),
+        ("fake-html.html", FileType.HTML),
+        ("stanley-cups.xlsx", FileType.XLSX),
+        ("stanley-cups.csv", FileType.CSV),
+        ("stanley-cups.tsv", FileType.TSV),
+        ("fake-power-point.pptx", FileType.PPTX),
+        ("winter-sports.epub", FileType.EPUB),
+        ("fake-doc.rtf", FileType.RTF),
+        ("spring-weather.html.json", FileType.JSON),
+        ("fake.odt", FileType.ODT),
+        ("fake-incomplete-json.txt", FileType.TXT),
+    ],
+)
+def test_detect_filetype_from_filename_extension_when_file_does_not_exist(
+    file_name: str, expected_value: FileType, monkeypatch: MonkeyPatch
+):
+    """File-type is detected from extension when file does not exist."""
```

Comment:
These are not new test cases, just separating `test_detect_filetype_from_filename_with_extension()` into the part that stays (works when libmagic not available) and the part that will be removed (derives `FileType` from extension when file does not exist at `file_path`).